source/common/upstream/eds.cc

```diff
@@ -39,7 +39,10 @@ EdsClusterImpl::EdsClusterImpl(const envoy::api::v2::Cluster& cluster, Runtime::
       "envoy.api.v2.EndpointDiscoveryService.StreamEndpoints");
 }
 
-void EdsClusterImpl::initialize() { subscription_->start({cluster_name_}, *this); }
+void EdsClusterImpl::initialize(std::function<void()> callback) {
+  setInitializeCallback(callback);
+  subscription_->start({cluster_name_}, *this);
```

Comment:
please comment on why we don't startInitialization() and that there won't be a finishInitialization in (at least) the DNS Cluster

I'm mildly inclined to say these should be overrides of startInitialization rather than initialize() as they all start with setInitializaeCallback but if you want to avoid the extra virtual function you're welcome to leave as is and just comment :-)

Reply:
Sorry I don't follow. There should always be a finishInititalization() for every cluster now, kicked off by startInitialization().

Sure, I can add the additional virtual function call.

Reply:
I think I misread what was being overridden and thought this was replacing not preempting the base class initialization.  More clear as it's written now -  I don't love the naming but can't think of anything better :-P

I believe the mac breakage is unrelated - is there another known flake there I should check out?

Reply:
Yeah I think there is still some OSX flake. :( cc @zuercher 

FWIW, I tried to think of better names for all of this but I can't think of anything better either. I hope the comments help.